network/p2pNetwork.go

```diff
@@ -560,7 +560,20 @@ func (n *P2PNetwork) OnNetworkAdvance() {}
 
 // GetHTTPRequestConnection returns the underlying connection for the given request. Note that the request must be the same
 // request that was provided to the http handler ( or provide a fallback Context() to that )
-func (n *P2PNetwork) GetHTTPRequestConnection(request *http.Request) (conn net.Conn) { return nil }
+func (n *P2PNetwork) GetHTTPRequestConnection(request *http.Request) (conn DeadlineSettable) {
+	addr := request.Context().Value(http.LocalAddrContextKey).(net.Addr)
+	peerID, err := peer.Decode(addr.String())
+	if err != nil {
+		n.log.Infof("GetHTTPRequestConnection failed to decode %s", addr.String())
+		return nil
+	}
+	conn, ok := n.service.GetStream(peerID)
+	if !ok {
+		n.log.Warnf("GetHTTPRequestConnection no such stream for peer %s", peerID.String())
+		return nil
```

Comment:
Do we want this function to return an error here perhaps?

Reply:
The original `wsNetwork.GetHTTPRequestConnection` could also return nil and the ledger service (client) aware. So it is safe. I warn here to emphasize it is not quite expected for p2p net but ok to proceed.